File: pulso/data/contagion.py
"""
Emotion contagion matrix for PULSO.
Defines how emotions spread from major nodes to satellite nodes.
Matches Section 5.3 of the master briefing exactly.
"""

CONTAGION_MATRIX = {
    "anger":   {"anger": 0.60, "fear": 0.20, "sadness": 0.10, "joy": 0.05, "hope": 0.05},
    "joy":     {"joy": 0.60, "hope": 0.20, "anger": 0.05, "fear": 0.05, "sadness": 0.10},
    "fear":    {"fear": 0.55, "sadness": 0.20, "anger": 0.15, "joy": 0.03, "hope": 0.07},
    "hope":    {"hope": 0.55, "joy": 0.25, "fear": 0.05, "sadness": 0.05, "anger": 0.10},
    "sadness": {"sadness": 0.55, "fear": 0.20, "anger": 0.10, "joy": 0.05, "hope": 0.10},
}

File: frontend/js/data/contagion.js
// Emotion contagion matrix — frontend visual distribution for satellite nodes.
// Values tuned for cluster COLOR DOMINANCE: 70% primary so each cluster reads
// as one dominant hue from a distance. Backend contagion.py uses original values.
//
// Distribution: primary 0.70, secondary 0.15, tertiary 0.08, others 0.035 each.
const CONTAGION_MATRIX = {
  anger:   { anger: 0.70, fear: 0.15, sadness: 0.08, joy: 0.035, hope: 0.035 },
  joy:     { joy:   0.70, hope: 0.15, sadness: 0.08, anger: 0.035, fear: 0.035 },
  fear:    { fear:  0.70, sadness: 0.15, anger: 0.08, hope: 0.035, joy: 0.035 },
  hope:    { hope:  0.70, joy:  0.15, anger: 0.08, fear: 0.035, sadness: 0.035 },
  sadness: { sadness: 0.70, fear: 0.15, anger: 0.08, joy: 0.035, hope: 0.035 },
};
